Index: systemtap/stap-prep
===================================================================
--- systemtap.orig/stap-prep
+++ systemtap/stap-prep
@@ -133,7 +133,11 @@ esac
     echo "linux-image-$ABINAME = $VERSION"
     echo "linux-headers-$ABINAME = $VERSION"
     if [ "$DEBUGINFOD" -eq 0 ]; then # add dbgsym search if not already 
-        echo "linux-image-$ABINAME-dbgsym = $VERSION"
+        if [ "$DISTRO" = "Debian" ]; then
+	        echo "linux-image-$ABINAME-dbg = $VERSION"
+        else
+            echo "linux-image-$ABINAME-dbgsym = $VERSION"
+        fi
     fi
 ) | while read package relation requiredversion; do
     installedversion="$(dpkg-query -W "$package" 2> /dev/null | cut -f 2)"
