Index: systemtap/includes/sys/sdt.h
===================================================================
--- systemtap.orig/includes/sys/sdt.h
+++ systemtap/includes/sys/sdt.h
@@ -100,6 +100,8 @@
 # ifndef STAP_SDT_ARG_CONSTRAINT
 # if defined __powerpc__
 # define STAP_SDT_ARG_CONSTRAINT        nZr
+# elif defined __ia64__
+# define STAP_SDT_ARG_CONSTRAINT        nr
 # elif defined __arm__
 # define STAP_SDT_ARG_CONSTRAINT        g
 # else
Index: systemtap/python/HelperSDT/_HelperSDT.c
===================================================================
--- systemtap.orig/python/HelperSDT/_HelperSDT.c
+++ systemtap/python/HelperSDT/_HelperSDT.c
@@ -347,7 +347,11 @@ init_HelperSDT(void)
 	// it with the asm() statement. Otherwise we get a @GOTPCREL
 	// reference which stap can't parse.
 	void *fptr = &PyObject_GenericGetAttr;
+#ifdef __ia64__
+	asm ("nop 0" : "=r"(fptr) : "r"(fptr));
+#else
 	asm ("nop" : "=r"(fptr) : "r"(fptr));
+#endif
 	STAP_PROBE2(PROVIDER, Init, stap_module, fptr);
     }
     return module;
